_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp

```diff
@@ -500,8 +500,8 @@ namespace UMC_MPEG2_DECODER
         }
 
         const auto picExt = *m_currHeaders.picExtHdr;
-        uint32_t fieldIndex = m_currFrame->GetNumberByParity(picExt.picture_structure == BOTTOM_FLD_PICTURE);
-        MPEG2DecoderFrameInfo & info = *m_currFrame->GetAU(fieldIndex);
+        m_currFrame->fieldIndex = m_currFrame->GetNumberByParity(picExt.picture_structure == BOTTOM_FLD_PICTURE);
```

Comment:
Can we do it only **once** on the first slice, not when adding **each new** slice?

Reply:
It doesn't seem feasible.
1. after `MPEG2DecoderFrameInfo & info = *m_currFrame->GetAU(fieldIndex)`, we can get the slice size according to `info.GetSliceCount()`
2. according to `slices.size() = 0`, we can get the first slice.
And there is a judgment of the first slice in [MPEG2DecoderFrameInfo::AddSlice](https://github.com/Intel-Media-SDK/MediaSDK/blob/master/_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_frame.cpp#L64)

Reply:
Sorry, I don't follow what is not feasible here. 
See https://github.com/Intel-Media-SDK/MediaSDK/blob/b4cfab66cfe92958fed6bff30691ed11d8748eb9/_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp#L723